modules/imgproc/src/approx.cpp

```diff
@@ -860,4 +861,239 @@ cvApproxPoly( const void* array, int header_size,
     return dst_seq;
 }
 
+struct neighbours
+{
+    /*
+        if relevant = -1, point has been removed
+        if relevant = 0,  the intersection in the heap must be recalculated
+        if relevant = 1,  intersection has been calculated
+    */
+    short relevant;
```

Comment:
use `int8_t`  instead of `short` or use `enum class`

Also suggest to rename `relevant` to `pointStatus`